Feature/config gain scheduling

Deakensbubbles opened the pull request:
<!--
To streamline the process of creating and reviewing pull requests, we have created a template.
It is not required to follow this template perfectly, but we encourage you to think about each header.
Before you publish a pull request think about the definition of done for the issue you are trying to resolve.
-->

<!--
Provide the key for the Jira issue this PR resolves.
-->

Closes PM-

## Description
<!--
Provide a concise description on the feature/fix your pull request implements.
-->
Added a config file containing all the PID values per gaittype. The Python Node now makes use of this file for determining the gain values. 
Fixed a naming issue in the march_gain_scheduling_node caused by the name changes in #382 .
## Changes
<!--
Provide a list of important changes.

e.g.
* Added tests
* Renamed variable
* Added topic
-->
Added a config folder and gaittype_gains.yaml file.
Changes in the lookup_table method in the dynamic_pid_reconfigurer.py file to use the config file.
Changes in march_gain_scheduling_node to use the correct python packages.
<!-- Please don't forget to request for reviews -->

Deakensbubbles commented on march_gain_scheduling/config/gaittype_gains.yaml:
```diff
@@ -0,0 +1,28 @@
+gaittypes: # The used gains are only for testing that the correct set of gains has been loaded.
```
I have thought about this namespacing, but the issue becomes apparent when considering the full name; gaittype gains. When read as "gait type gains" it is unclear whether the word type applies to gains or to gaits. Therefore, though it might be slightly harder to read, it is clearer that we are talking about gains that correspond to a certain gait-type.

Olavhaasie replied:
To me it is clear enough since `type` seems more common to apply to `gait` than to `gain`. Moreover the parameters are under `/gait_type/<gait>_gains`, which also makes it clear. The only problem is then the file name, so you could call it just `gains.yaml`?

Deakensbubbles replied:
Yes, it more commonly applies to gait, but I doubt that March VI would think about this the same way as we do. If you have to check the parameters to ascertain if the namespace is talking about a type of gains or a type of gait, then you have an incomprehensible name. To me it should be clear from reading the name of the config file what I can find inside it. Naming the config file `gains.yaml` would be way too arbitrary in my opinion.

Roelemans replied:
Maybe name it gains_per_gait_type?

Deakensbubbles replied:
sure, I can work with that!

Deakensbubbles commented on march_gain_scheduling/src/march_gain_scheduling/dynamic_pid_reconfigurer.py:
```diff
@@ -24,30 +24,16 @@ def gait_selection_callback(self, data):
-            return 10007, 0, 10
-        elif self._joint_list[i] == "right_knee" and self._gait_name == "sit_down":
-            return 10008, 0, 10
+        if rospy.has_param("/gaittypes/"+self._gait_name+"_gains/"):
```
I will add the spaces between the operators. Funnily enough CLion was pestering me that there shouldn't be spaces. I think it might be a conflict of conventions. Please read my other comment concerning the gaittype namspacing.

Olavhaasie replied:
:thinking: My CLion does not give me that warning. We should take a look at CLion configurations, since mine tells me to add spaces according to PEP8.

Deakensbubbles commented:
Thank you @Olavhaasie for the recommendation of the launch file. I will add this later on in another pull request, perhaps along with a few other changes concerning the linear transition between the gain values.

Deakensbubbles commented:
added the argument as you suggested @Olavhaasie and removed the unnecessary launch files. I have also renamed the march4 folder in the config folder to march-iv in accordance with all the other folders in the workspace.

Olavhaasie commented:
Actually, we have just renamed the URDF and launch files to use `march4` to be more consistent with the hardware interface. If you could point out anything that is still named `march-iv`, then we can change that as well. 

So if you could revert the rename, then everything’s 👌

Deakensbubbles commented:
The pull request should now be according to our standards
